src/ast/helpers/ignore_field.rs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use core::cmp::{Eq, Ord, Ordering, PartialEq, PartialOrd};
+use core::fmt::{self, Debug, Formatter};
+use core::hash::{Hash, Hasher};
+
+/// A wrapper type that ignores the field when comparing or hashing.
+pub struct IgnoreField<T>(pub T);
```

Comment:
Instead of using a generic type would it make sense to represent via a regular wrapper type (thinking since the plan is only to wrap the tokens we attach to the nodes)? e.g.
```rust
pub struct AttachedToken(TokenWithLocation);

impl From<Token> for AttachedToken {}
impl From<TokenWithLocation> for AttachedToken {}
impl Hash for AttachedToken {}
// maybe even a default/noop impl to make it easier in scenarios like test cases where the info is irrelevant
// etc.

pub struct With {
    pub with_token: AttachedToken,
    pub recursive: bool,
    // ...
}
```

Reply:
doneso 👍